### DISCUSSION TITLE
why not generate only a header file from proto files?

### DISCUSSION THREAD

**Discussion by CheyenneForbes** [protobuf-c/protobuf-c#641] [Q&A] [answered]
Maybe a c and h file is generated for some good reason (well, it is because thats how it was coded to be done but you know what I mean). I was thinking, why not just have a .h file that I can just include in my c file?

**Comment by FSMaxB** [comment]
I don't know why the decision was actually made, but one thing that comes to mind is that you only have to compile the C-File once. If you put the code in the header, you have to compile the generated code as many times as you have included the header. And that can be a lot of files.

**Comment by edmonds** [comment, marked as answer]
The `.pb-c.c` file contains static data which can be reused inside a project by using the linker. This is no different than the C++ protobuf project's `protoc` which generates `.pb.h` and `.pb.cc` files for similar reasons. Possibly you could include both the `.pb-c.h` and `.pb-c.c` files if you are only using the generated files from a single C file.
